src/pages/recipes/detail.js

```diff
@@ -0,0 +1,112 @@
+import {
+	ClockIcon,
+	BellIcon,
+	LinkIcon,
+	FireIcon,
+} from "@heroicons/react/24/outline";
+import { useEffect } from "react";
+import Image from "next/image";
+
+const Modal = ({ isOpen, onClose, recipe }) => {
+	useEffect(() => {
+		if (isOpen) {
+			const handleEsc = (event) => {
+				if (event.key === "Escape") {
+					onClose();
+				}
+			};
+			window.addEventListener("keydown", handleEsc);
+			return () => {
+				window.removeEventListener("keydown", handleEsc);
+			};
+		}
+	}, [isOpen, onClose]);
+
+	if (!isOpen || !recipe) return null;
+
+	return (
+		<div
+			className="fixed inset-0 bg-gray-900 bg-opacity-90 flex items-center justify-center z-50 p-4 sm:p-8"
+			role="dialog"
+			aria-modal="true"
+		>
+			<div className="bg-gray-800 text-gray-200 rounded-lg max-w-full sm:max-w-lg w-full max-h-[90vh] overflow-y-auto relative shadow-xl p-6">
+				<button
+					className="absolute top-2 right-4 text-gray-300 hover:text-gray-100 transition duration-300 text-2xl"
+					onClick={onClose}
+					aria-label="Close modal"
+				>
+					&times;
+				</button>
```

Comment:
hi @SeesonLau, can you make the `x button` sticky when scrolling down the modal?